src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadonlyStructTests.cs

```diff
@@ -2136,5 +2152,171 @@ .locals init (S1 V_0)
             // should warn about calling s2.M2 in warning wave (see https://github.com/dotnet/roslyn/issues/33968)
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void ReadOnlyMethod_OverrideBaseMethod()
+        {
+            var csharp = @"
+public struct S1
+{
+    // note: GetType can't be overridden
+    public override string ToString() => throw null;
+    public override int GetHashCode() => throw null;
+    public override bool Equals(object o) => throw null;
+
+    readonly void M()
+    {
+        // should warn--non-readonly invocation
+        ToString();
+        GetHashCode();
+        Equals(null);
+
+        // ok
+        base.ToString();
+        base.GetHashCode();
+        base.Equals(null);
+    }
+}
+
+public struct S2
+{
+    public readonly override string ToString() => throw null;
+    public readonly override int GetHashCode() => throw null;
+    public readonly override bool Equals(object o) => throw null;
+
+    readonly void M()
+    {
+        // no warnings
+        ToString();
+        GetHashCode();
+        Equals(null);
+
+        base.ToString();
+        base.GetHashCode();
+        base.Equals(null);
+    }
+}
+";
+            var verifier = CompileAndVerify(csharp);
+            verifier.VerifyDiagnostics(
+                // (12,9): warning CS8655: Call to non-readonly member 'ToString' from a 'readonly' member results in an implicit copy of 'this'.
+                //         ToString();
+                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "ToString").WithArguments("ToString", "this").WithLocation(12, 9),
+                // (13,9): warning CS8655: Call to non-readonly member 'GetHashCode' from a 'readonly' member results in an implicit copy of 'this'.
+                //         GetHashCode();
+                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "GetHashCode").WithArguments("GetHashCode", "this").WithLocation(13, 9),
+                // (14,9): warning CS8655: Call to non-readonly member 'Equals' from a 'readonly' member results in an implicit copy of 'this'.
+                //         Equals(null);
+                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "Equals").WithArguments("Equals", "this").WithLocation(14, 9));
+
+            verifier.VerifyIL("S1.M", @"
```

Comment:
>VerifyIL [](start = 21, length = 8)

nit: consider adding a comment to point out the important thing you're trying to verify in the IL

Reply:
Will do. Will also split this test into two and migrate one more base method test over from ReadOnlyStructsTests. Should help with readability.

---
In reply to: [270201319](https://github.com/dotnet/roslyn/pull/34468#discussion_r270201319) [](ancestors = 270201319)

Reply:
nit: I'm not sure the split between two test files helps (whenever a test moves, the review becomes more confusing).

---
In reply to: [270204398](https://github.com/dotnet/roslyn/pull/34468#discussion_r270204398) [](ancestors = 270204398,270201319)

Reply:
Ah, do you think I should just leave it in its original file even though it's now verifying IL? I'm fine with that.